src/tribler-common/tribler_common/sentry_reporter/sentry_tools.py

```diff
@@ -0,0 +1,125 @@
+""" This a collection of tools for SentryReporter and SentryScrubber aimed to
+simplify work with several data structures.
+"""
+from collections import defaultdict
+
+
+def parse_os_environ(array):
+    """ Parse os.environ field.
+
+    Args:
+        array: strings that represents tuples delimited by `:`
+            Example: ["KEY:VALUE", "PATH:~/"]
+
+    Returns:
+        Dictionary that made from given array.
+            Example: {"KEY": "VALUE", "PATH": "~/"}
+
+    """
+    result = defaultdict(lambda: '')
+
+    if not array:
+        return result
+
+    for line in array:
+        items = line.split(':', 1)
+        if len(items) < 2:
+            continue
+        result[items[0]] = items[1]
+
+    return result
+
+
+def parse_stacktrace(stacktrace):
+    """ Parse stacktrace field.
+
+    Args:
+        stacktrace: a string with '\n' delimiter.
+            Example: "line1\nline2"
+
+    Returns:
+        The list of strings made from the given string.
+            Example: ["line1", "line2"]
+    """
+    if not stacktrace:
+        return []
+
+    return [line for line in stacktrace.split('\n') if line]
+
+
+def first(l, default=''):
+    """ Safe return the first item in a list.
+
+    Args:
+        l: a list
+        default: a default value to return in case no items in list.
+
+    Returns:
+        The first item in list.
+    """
+    return l[0] if l else default
+
+
+def last(l, default=''):
```

Comment:
Personally, I think adding 9 lines of documentation for a trivial one-liner is clear overkill, and actually *decreases* readability. I found it much easier to understand what the function does just by looking at the code. Having to parse through the comments required some effort from me.

Also, the variable is named `l`. People don't use `l` as a variable name for a reason (there is even a pylint check for that).

Reply:
What can you offer instead?

Reply:
`x`?

Reply:
Nice! `iterable` looks ok, will change the name.

Reply:
Fixed.